estimagic/tests/dashboard/test_monitoring_app.py

```diff
@@ -75,85 +84,112 @@ def fake_read_last_rows(**kwargs):
     assert res == 0
 
 
+def test_create_id_column_single_index():
+    start_params = pd.DataFrame()
+    start_params["value"] = [1, 2, 3, 4]
+    start_params["group"] = ["a", "a", "b", "b"]
+    start_params["name"] = ["this", "repeats"] * 2
+    start_params.index = [2, 3, 4, 5]
+
+    res = monitoring._create_id_column(start_params)
+    expected = pd.Series(list("2345"), index=start_params.index)
+    pdt.assert_series_equal(res, expected)
+
+
+def test_create_id_column_multi_index():
+    multi_params = pd.DataFrame()
+    multi_params["value"] = [1, 2, 3, 4]
+    multi_params["group"] = ["a", "a", "b", "b"]
+    multi_params["name"] = ["this", "repeats"] * 2
+    multi_params["3rd level"] = [3, 4, 5, 6]
+    multi_params.set_index(["group", "name", "3rd level"], inplace=True)
+
+    res = monitoring._create_id_column(multi_params)
+    expected = pd.Series(
+        ["a_this_3", "a_repeats_4", "b_this_5", "b_repeats_6"], index=multi_params.index
+    )
+    pdt.assert_series_equal(res, expected)
```

Comment:
The distinction between the two cases becomes irrelevant if we move to the proposed solution.

Reply:
Yes, that simplified the tests a lot. I parametrized the tests and kept the MultiIndex case in since I switched to using `groupby`. (b5fcf5d)